### PR TITLE
Remove zlib from libcudf public CMake dependencies

### PR DESCRIPTION
**Pull request by robertmaynard**
libcudf doesn't expose zlib in the public facing API, and therefore C++ consumers don't need to also link / include zlib.

### CODE REVIEW

**Comment by robertmaynard** [comment]
@kkraus14 Followup from your comment about `ZLIB::ZLIB`

**Review by kkraus14 — APPROVED** [review]

**Comment by robertmaynard** [comment]
rerun tests

**Review by harrism — APPROVED** [review]

File: cpp/CMakeLists.txt [hunk 1]
```diff
@@ -506,11 +505,11 @@ add_dependencies(cudf jitify_preprocess_run)
 
 # Specify the target module library dependencies
 target_link_libraries(cudf
-           PUBLIC ZLIB::ZLIB
-                  ${ARROW_LIBRARIES}
+           PUBLIC ${ARROW_LIBRARIES}
                   cudf::Thrust
                   rmm::rmm
            PRIVATE cuco::cuco
+	                 ZLIB::ZLIB
```
Comment (kkraus14) [inline cpp/CMakeLists.txt:512]:
```suggestion
	             ZLIB::ZLIB
```

**Review by kkraus14 — APPROVED** [review]

**Comment by kkraus14** [comment]
@gpucibot merge

**Comment by robertmaynard** [comment]
@gpucibot merge

**Pull request merged**